src/main/java/com/groupthree/culinarycompanion/dao/UserDAOStub.java

```diff
@@ -7,40 +7,76 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * An implementation of IUserDAO that uses an in-memory database represented by a List of User objects.
+ */
 @Repository
 public class UserDAOStub implements IUserDAO {
-    private List<User> userDatabase = new ArrayList<>();
+    private final List<User> userDatabase = new ArrayList<>();
     private int nextUserId = 1;
 
+    /**
+     * Finds user by their unique identifier.
+     *
+     * @param userId The unique identifier of the user.
+     * @return The user with the specified ID.
+     * @throws UserNotFoundException if no user with the specified ID is found.
+     */
     @Override
     public User findUserById(int userId) {
         for (User user : userDatabase) {
             if (user.getUserId() == userId) {
                 return user;
             }
         }
-        return null;
+        throw new UserNotFoundException("User with ID" + userId + "not found");
```

Comment:
This is a suggestion for enhancing the null handling of this class, doing so ensures there is no chance of a NullPointerException. Cited from: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3